packages/nuxt/src/core/templates.ts

```diff
@@ -374,10 +374,13 @@ export const nuxtConfigTemplate: NuxtTemplate = {
       baseURL: undefined,
       headers: undefined
     }
+    const shouldEnableComponentIslands = ctx.nuxt.options.experimental.componentIslands && (
+      ctx.nuxt.options.dev || ctx.nuxt.options.experimental.componentIslands !== 'auto' || ctx.app.pages?.some(p => p.mode === 'server') || ctx.app.components?.some(c => c.mode === 'server')
```

Comment:
`ctx.nuxt.options.experimental.componentIslands !== 'auto'` we can set `false` to `componentIsland`, not sure if anyone does that but it can enable island

Reply:
I think the first check prevents that - `ctx.nuxt.options.experimental.componentIslands &&`

Reply:
oh indeed !